tokio/tests/task_local_set.rs

```diff
@@ -0,0 +1,50 @@
+#![warn(rust_2018_idioms)]
+#![cfg(feature = "full")]
+
+use tokio::runtime::Runtime;
+use tokio::sync::oneshot;
+use tokio::task::{self, LocalSet};
+
+#[test]
+fn acquire_mutex_in_drop() {
```

Comment:
The failure mode for this is a deadlock, right? Should we maybe do something similar to this test and add a timeout? https://github.com/tokio-rs/tokio/blob/efb4b67a5415c1feeaa6668618243c89f7b601eb/tokio/src/task/local.rs#L732-L741

Reply:
I rather it just deadlock tbh... timeouts don't seem great.